frontend/apps/remark42/app/common/cookies.ts

```diff
@@ -27,15 +26,66 @@ export function setCookie(name: string, value: string, options: CookieOptions =
   let updatedCookie = `${name}=${value}`;
 
   for (const [key, value] of Object.entries(options)) {
-    updatedCookie += `; ${key}`;
-    if (value !== true) {
-      updatedCookie += `=${value}`;
+    // For boolean attributes like 'secure', only add them if true, otherwise skip
+    if (typeof value === 'boolean') {
+      if (value) {
+        updatedCookie += `; ${key}`;
+      }
+      // Skip false boolean values entirely
+    } else if (value !== undefined) {
+      // For non-boolean values
+      updatedCookie += `; ${key}=${value}`;
     }
   }
 
   document.cookie = updatedCookie;
 }
 
+/**
+ * Sets a cookie with enhanced security options for authentication
+ * @param name The name of the cookie
+ * @param value The value to set
+ * @param options Additional cookie options
+ */
+export function setAuthCookie(name: string, value: string, options: CookieOptions = {}) {
+  const isSecure = window.location.protocol === 'https:';
+  const cookiePrefix = isSecure ? '__Host-' : '';
+
+  // Default options for auth cookies with strong security
+  const authOptions: CookieOptions = {
+    path: '/',
+    sameSite: 'Strict',
+    secure: isSecure,
+    ...options,
+  };
+
+  setCookie(`${cookiePrefix}${name}`, value, authOptions);
+}
+
+/**
+ * Clears an authentication cookie by setting its expiration to the past
+ * @param name The name of the cookie to clear
+ */
+export function clearAuthCookie(name: string) {
+  const isSecure = window.location.protocol === 'https:';
+  const cookiePrefix = isSecure ? '__Host-' : '';
+
+  setCookie(`${cookiePrefix}${name}`, '', {
+    path: '/',
+    secure: isSecure,
+    expires: new Date(0), // Set to epoch time to expire immediately
+  });
+
+  // Also try to clear the non-prefixed version to be thorough
+  if (cookiePrefix) {
+    setCookie(name, '', {
+      path: '/',
+      secure: isSecure,
+      expires: new Date(0),
+    });
+  }
```

Comment:
these 2 calls are basically identical, the only difference is the name.
Consider calling the function once with different cookie name passed.

Reply:
I don't get you.

Reply:
```ts
setCookie(`${cookiePrefix ?? ""}${name}`, '', {
    path: '/',
    secure: isSecure,
    expires: new Date(0), // Set to epoch time to expire immediately
  });
```

Reply:
I'm cleaning up both here, but the code you're proposing cleaning only one variant.